tool/net/redbean.c

```diff
@@ -6912,6 +6924,7 @@ static int HandlePoll(int ms) {
         if (!polls[pollid].revents) continue;
         if (polls[pollid].fd < 0) continue;
         if (polls[pollid].fd) {
+          if (maxworkers && shared->workers >= maxworkers) continue;
```

Comment:
Shouldn't you accept the connection and then send like a 503 Service Unavailable?

Reply:
I don't think it's needed, as a worker may become available. It's not much different from how uniprocess works, just for `maxworkers` instead of 1.

Reply:
Well if the event isn't handled, then won't poll spin and consume all the cpu?

Reply:
It shouldn't, as the effect is going to be the same as if the event wasn't there in the first place.

Think about it this way: if 5 requests come in and we are running redbean with 1 worker (uniprocess=true), the requests will be queued and handled one by one. If we're running redbean with 4 workers (maxworkers=4), only one requests will be queued and handled when one of the workers becomes available or is finished (to allow a new worker to be launched).

Since `poll` is not going to block (as there is an event pending), `HandlePoll` will exit in this case and the rest of the processing will be handled normally (inside `EventLoop`), including `ReapZombies` call, which includes handling of exited workers (`HandleWorkerExit`). Then the loop will repeat and if the number of workers goes down, a new worker will be launched.

Or am I missing something?

Reply:
I'm going to merge it and see if I can confirm what I suspect will happen actually happens.